examples/message_broker.rs

```diff
@@ -7,11 +7,11 @@ use zeromq::{Socket, ZmqError};
 async fn main() -> Result<(), Box<dyn Error>> {
     let mut frontend = zeromq::RouterSocket::new();
     frontend
-        .bind("127.0.0.1:5559")
+        .bind("tcp://127.0.0.1:5559".parse()?)
```

Comment:
Imho this is not the best solution. I try to keep public api simple as possible and here you force user to make 2 error checks instead of one. bind() itself may return an error and IMHO we should just accept &str and do the parsing inside. In such case we can return appropriate error from bind call in case of malformed address

Reply:
Would the impl Trait approach solve that? See https://github.com/zeromq/zmq.rs/pull/68/commits/15a3c814e248c8b06f3bb5c8fae6228ce9effa63. In that version, you can simply provide "tcp://127.0.0.1:5559" without a parse or `?` operator

Reply:
Yes I think that would be a better option for now

Reply:
I just added back in the `impl Trait` approach. Does that resolve this issue?